src/assets/wise5/classroomMonitor/student-progress/student-progress.component.ts

```diff
@@ -0,0 +1,252 @@
+      this.sortedTeams.push(team);
+    }
+    switch (this.sort) {
+      case 'team':
+        this.sortedTeams.sort(this.createSortTeam('asc'));
+        break;
+      case '-team':
+        this.sortedTeams.sort(this.createSortTeam('desc'));
+        break;
+      case 'student':
+        this.sortedTeams.sort(this.createSortStudent('asc'));
+        break;
+      case '-student':
+        this.sortedTeams.sort(this.createSortStudent('desc'));
+        break;
+      case 'score':
+        this.sortedTeams.sort(this.createSortScore('asc'));
+        break;
+      case '-score':
+        this.sortedTeams.sort(this.createSortScore('desc'));
+        break;
+      case 'completion':
+        this.sortedTeams.sort(this.createSortCompletion('asc'));
+        break;
+      case '-completion':
+        this.sortedTeams.sort(this.createSortCompletion('desc'));
+        break;
+      case 'location':
+        this.sortedTeams.sort(this.createSortLocation('asc'));
+        break;
+      case '-location':
+        this.sortedTeams.sort(this.createSortLocation('desc'));
+        break;
+    }
+  }
+
+  private createSortTeam(direction: string): any {
+    return (workgroupA: any, workgroupB: any): number => {
+      if (direction === 'asc') {
+        return workgroupA.workgroupId - workgroupB.workgroupId;
+      } else {
+        return workgroupB.workgroupId - workgroupA.workgroupId;
+      }
+    };
+  }
+
+  private createSortStudent(direction: string): any {
+    return (workgroupA: any, workgroupB: any): number => {
+      let localeCompare: number;
+      if (direction === 'asc') {
+        localeCompare = workgroupA.username.localeCompare(workgroupB.username);
+      } else {
+        localeCompare = workgroupB.username.localeCompare(workgroupA.username);
+      }
+      if (localeCompare === 0) {
+        return workgroupA.workgroupId - workgroupB.workgroupId;
+      } else {
+        return localeCompare;
+      }
+    };
+  }
+
+  private createSortScore(direction: string): any {
+    return (workgroupA: any, workgroupB: any): number => {
+      if (workgroupA.scorePct === workgroupB.scorePct) {
+        return workgroupA.username.localeCompare(workgroupB.username);
+      } else {
+        if (direction === 'asc') {
+          return workgroupA.scorePct - workgroupB.scorePct;
+        } else {
+          return workgroupB.scorePct - workgroupA.scorePct;
+        }
+      }
+    };
+  }
+
+  private createSortCompletion(direction: string): any {
+    return (workgroupA: any, workgroupB: any): number => {
+      const completionA = workgroupA.completion.completionPct;
+      const completionB = workgroupB.completion.completionPct;
+      if (completionA === completionB) {
+        return workgroupA.username.localeCompare(workgroupB.username);
+      } else {
+        if (direction === 'asc') {
+          return completionA - completionB;
+        } else {
+          return completionB - completionA;
+        }
+      }
```

Comment:
The if-else cases in the sort functions can be refactored a bit to reduce complexity. For example:
```suggestion
      if (completionA === completionB) {
        return workgroupA.username.localeCompare(workgroupB.username);
      }
      return direction === 'asc' ? completionA - completionB : completionB - completionA;
```